### ISSUE TITLE
Crash when connecting to server

### ISSUE THREAD

**Issue by Deathsbreed** [minetest/minetest#4867]
I was trying to connect to a server "Total Chaos - NO RULES" and my game crashed giving this error:
```
minetest: /build/minetest-UurSbT/minetest-0.4.14+repack/src/nodedef.cpp:938: virtual void CNodeDefManager::updateTextures(IGameDef*, void (*)(void*, irr::u32, irr::u32), void*): Assertion `f->liquid_type == LIQUID_SOURCE' failed.
```
Before it there were a bunch of other warnings like this:
```
2016-12-08 16:24:14: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:14: WARNING[Main]: Irrlicht: PNG warning: bKGD: invalid
2016-12-08 16:24:14: WARNING[Main]: Irrlicht: PNG warning: bKGD: invalid
2016-12-08 16:24:14: WARNING[Main]: Irrlicht: PNG warning: bKGD: invalid
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: Interlace handling should be turned on when using png_read_image
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: Interlace handling should be turned on when using png_read_image
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: Interlace handling should be turned on when using png_read_image
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: Interlace handling should be turned on when using png_read_image
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: Interlace handling should be turned on when using png_read_image
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:15: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: profile 'Photoshop ICC profile': 'RGB ': RGB color space not permitted on grayscale PNG
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: iCCP: known incorrect sRGB profile
2016-12-08 16:24:16: WARNING[Main]: Irrlicht: PNG warning: Interlace handling should be turned on when using png_read_image
```
I'm running the Debian Stretch packaged version of Minetest (0.4.14+repack-5).

The screen shows "Initializing nodes 3%..." before crashing.

**Comment by rubenwardy** [comment]
Can you reproduce this? Or was it one off

**Comment by Deathsbreed** [comment]
Every time I connect it happens, so I can reproduce it on my end.

**Comment by Fixer-007** [comment]
Can't reproduce on newest minetest. Connected fine.

**Comment by Deathsbreed** [comment]
I took a screencast of it just now showing the problem, you can find the video [here](https://goblinrefuge.com/mediagoblin/u/deathsbreed/m/minetest-bug-error-4867/). MediaGoblin tends to lower the quality of videos for their web player so you might want to download the original MP4 file.

**Comment by nerzhul** [comment]
@Deathsbreed can you test it on current master ?

**Comment by Deathsbreed** [comment]
@nerzhul, it seems as though the server that was causing the issue is no longer up. I tried connecting to the domain and port directly (just in case they simply changed names) and it's not there (this issue only occured with that server).

However, I would point out that I am no longer on the same system, as I switched from Debian to Arch, so I am unsure if it would yield the same results.

Because it seems like this has only ever affected me and only on one server, I'll close the issue until it pops up again (if it does).
